packages/HilltopCrawler/src/main/kotlin/nz/govt/eop/hilltop_crawler/fetcher/TaskMappers.kt

```diff
@@ -170,7 +190,43 @@ class MeasurementDataTaskMapper(
     parsedContent: HilltopMeasurementValues
 ) :
     TaskMapper<HilltopMeasurementValues>(
-        HilltopMessageType.MEASUREMENT_DATA,
+        HilltopFetchTaskType.MEASUREMENT_DATA,
+        sourceConfig,
+        fetchedUri,
+        fetchedAt,
+        content,
+        parsedContent) {
+  override fun buildNewTasksList(): List<DB.HilltopFetchTaskCreate> = emptyList()
+
+  override fun buildKafkaMessage(): HilltopMessage? =
+      if (parsedContent.measurement != null) {
+        HilltopMeasurementsMessage(
+            sourceConfig.councilId,
+            baseUri,
+            fetchedAt,
+            parsedContent.measurement.siteName,
+            parsedContent.measurement.dataSource.measurementName,
+            parsedContent.measurement.data.values.first().timestamp.let { YearMonth.from(it) },
+            fetchedUri.toASCIIString(),
+            content,
+        )
+      } else {
+        null
+      }
+
+  override fun determineNextFetchAt(): Instant =
+      randomTimeBetween(fetchedAt.plus(20, ChronoUnit.DAYS), fetchedAt.plus(30, ChronoUnit.DAYS))
+}
+
```

Comment:
The logic that was previously here, has now moved into the `MeasurementDataLatestTaskMapper` and this one has been simplified to this.